### PR TITLE
pybind/ceph_argparse: cleanups preparing for type annotations

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/ceph_argparse.py [hunk 1]
```diff
@@ -127,7 +127,7 @@ def __init__(self, **kwargs):
         set any per-instance validation parameters here
         from kwargs (fixed string sets, integer ranges, etc)
         """
-        pass
+        self.typeargs = None
```
Comment (sebastian-philipp) [inline src/pybind/ceph_argparse.py:130]:
```suggestion
        self.typeargs: Optional[str] = None
```

**Review by sebastian-philipp — APPROVED** [review]

File: src/pybind/ceph_argparse.py (hunk 1 above)
Comment (tchaikov) [reply src/pybind/ceph_argparse.py:130]: thank you for the suggestion. after revisiting the code, i think i should drop this member variable in a follow-up commit. as it's never assigned.

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2021-06-18_12:04:35-rados-wip-kefu-testing-2021-06-18-1414-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2021-06-19_08:35:14-rados-wip-kefu-testing-2021-06-18-1414-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2021-06-19_11:27:01-rados-wip-kefu-testing-2021-06-19-1642-distro-basic-smithi/

i don't think the failures were related to this change.

as i reran the failed tests without this change:

- https://pulpito.ceph.com/kchai-2021-06-19_14:17:34-rados-wip-kefu-testing2-2021-06-19-2054-distro-basic-smithi/

File: src/pybind/ceph_argparse.py (hunk 1 above)
Comment (tchaikov) [reply src/pybind/ceph_argparse.py:130]: #42043

**Pull request merged**